src/Elmish/Hooks/UseEffect.purs

```diff
@@ -5,19 +5,30 @@ module Elmish.Hooks.UseEffect
 import Prelude
 
 import Effect.Aff (Aff)
-import Elmish (ReactElement, forkVoid)
-import Elmish.Component (ComponentName, wrapWithLocalState)
+import Elmish (forkVoid)
+import Elmish.Component (wrapWithLocalState)
 import Elmish.Hooks.Type (Hook, HookName, mkHook)
 
-useEffect' :: ComponentName -> { init :: Aff Unit, render :: Unit -> ReactElement } -> ReactElement
-useEffect' name = wrapWithLocalState name \{ init, render } ->
-  { init: forkVoid init
-  , update: \_ msg -> absurd msg
-  , view: \_ _ -> render unit
-  }
-
 -- | The `useEffect` hook takes an effect (`Aff`) to run and runs it in the
--- | `init` of the resulting component.
+-- | `init` of the resulting component. E.g.:
+-- |
+-- | ```purs
+-- | todos :: ReactElement
+-- | todos = withHooks do
+-- |   todos /\ setTodos <- useState (HookName "TodoState") []
+-- |
+-- |   useEffect (HookName "FetchTodos") do
+-- |     todos <- API.fetchTodos
+-- |     liftEffect $ setTodos todos
+-- |
+-- |   pure $ H.fragment $ todoView <$> todos
+-- | ```
 useEffect :: HookName -> Aff Unit -> Hook Unit
-useEffect name init =
-  mkHook name \n render -> useEffect' n { init, render }
+useEffect name initAff =
+  mkHook name \n render -> useEffect' n { init: initAff, render }
+  where
+    useEffect' n = wrapWithLocalState n \{ init, render } ->
+      { init: forkVoid init
+      , update: \_ msg -> absurd msg
+      , view: \_ _ -> render unit
+      }
```

Comment:
It doesn't really need to pass `initAff` as a parameter, does it? And once you get rid of that, the only remaining parameter is `render`, and now you can eta-reduce:

```suggestion
  mkHook name useEffect'
  where
    useEffect' n = wrapWithLocalState n \render ->
      { init: forkVoid initAff
      , update: \_ msg -> absurd msg
      , view: \_ _ -> render unit
      }
```

But better yet, `useEffect'` could just be inline:

```suggestion
  mkHook name \n -> wrapWithLocalState n \render ->
      { init: forkVoid initAff
      , update: \_ msg -> absurd msg
      , view: \_ _ -> render unit
      }
```

Reply:
Oh right!

Reply:
Well after seeing this, I got to thinking. It doesn’t make much sense to pretend that `mkHook` is more general than it is. The callback currently accepts a `ComponentName`, which really will only get passed to `wrapWithLocalState`. If there’s ever a use case for using some other method of creating an element, we can think about refactoring then. But for now, this is a much simpler API.

Reply:
Yeah, makes sense to me